src/client/mod.rs

```diff
@@ -82,6 +82,12 @@ struct ClientInner {
     session_pool: ServerSessionPool,
 }
 
+impl Drop for ClientInner {
```

Comment:
We mark the topology as closed when the ClientInner is dropped rather than the Client since there can be multiple copies of the Client